ydb/docs/en/core/reference/ydb-sdk/topic.md

```diff
@@ -1319,12 +1459,126 @@ When reading starts, the client code must transmit the starting consumer offset
 
 - Java
 
-  Setting the starting offset for reading is not supported in the current state of Java SDK.
+  Starting offset for reading in Java can be set only for AsyncReader.
+  In `StartPartitionSessionEvent` callback `StartPartitionSessionSettings` with desired ReadOffset can be passed to `confirm` method.
+  Offset that should be considered as committed can be set with `setCommittedOffset`.
+
+  ```java
+  @Override
+  public void onStartPartitionSession(StartPartitionSessionEvent event) {
+      event.confirm(StartPartitionSessionSettings.newBuilder()
+              .setReadOffset(lastReadOffset)
+              .setCommitOffset(lastCommitOffset)
```

Comment:
it's not clear from this example what data type should be used for these offsets